lib/errors.ts

```diff
@@ -14,18 +14,98 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import errors = require('resin-cli-errors');
-import patterns = require('./utils/patterns');
+import _ = require('lodash');
```

Comment:
Why not use `import { isEmpty } from 'lodash'` in order to reduce bundle size?

Reply:
Just wondering since we do this in various places while using only 1-2 lodash functions.

Reply:
Mainly because we don't do any bundling in the CLI :smiley:. The normal build is just raw JS, and although the standalone build bundles into a single executable, I believe it just works file by file. It's not clever enough for tree shaking, so requiring individual methods won't make any difference.

We have talked about using webpack here too, but it's not the top priority, so there hasn't been any movement there yet. Download size for the CLI would be nice to improve (mainly for workshops), but outside the web isn't not as important, and the standalone build provides a lot of similar benefits, and lots of others.

We could also potentially move to the tinier standalone lodash modules like https://www.npmjs.com/package/lodash.isempty, which would help overall install size, but I think they'd actually hurt startup time (I believe, not 100% sure, that the IO time reading modules from disk is the expensive bit, and 1 large file is typically cheaper than many small ones), and we've got lots of subdependencies that use lodash anyway, so it wouldn't help much.

Reply:
Thanks for the explanation. I forgot that CLI doesn't use Webpack and rushed to edit my comment but you beat me to it!